source/h2_decoder.c

```diff
@@ -384,6 +387,21 @@ static void s_decoder_run_state(
     }
 }
 
+static void s_decoder_run_state(
+    struct aws_h2_decoder *decoder,
+    const struct decoder_state *state,
+    struct aws_byte_cursor *input) {
+
+    s_decoder_run_state_ex(decoder, state, input, false /*preserve_scratch*/);
+}
+static void s_decoder_run_state_preserve_scratch(
```

Comment:
is there a better name we could use here, maybe ```_persistent``` or ```_continue``` or ```_stream``` or just something more indicative of the actual semantics here?

Reply:
I feel like, for this internal function, being explicit about the exact trick I'm pulling here is more helpful than referring to it with an abstract term. If we did this all the time I might invent terminology for it and use that terminology consistently